src/D2L.Bmx/ConsolePrompter.cs

```diff
@@ -59,38 +64,68 @@ internal class ConsolePrompter : IConsolePrompter {
 	}
 
 	string IConsolePrompter.PromptPassword() {
+		Func<char> readKey;
+		if( IS_WINDOWS ) {
+			// On Windows, Console.ReadKey calls native console API, and will fail without a console attached
+			if( Console.IsInputRedirected ) {
+				Console.Error.WriteLine( """
+					====== WARNING ======
+					Input to BMX is redirected. Password input may be displayed on screen!
+					If you're using mintty (with Git Bash, Cygwin, MSYS2 etc.), consider switching
+					to Windows Terminal for a better experience.
+					If you must use mintty, prefix your bmx command with 'winpty '.
+					=====================
+					""" );
+				readKey = () => (char)_stdinReader.Read();
+			} else {
+				readKey = () => Console.ReadKey( intercept: true ).KeyChar;
+			}
+		} else {
+			readKey = () => (char)_stdinReader.Read();
+		}
+
 		Console.Error.Write( "Okta Password: " );
 
-		if( RuntimeInformation.IsOSPlatform( OSPlatform.Windows ) ) {
-			var passwordChars = new List<char>();
+		string? originalTerminalSettings = null;
+		try {
+			if( !IS_WINDOWS ) {
+				originalTerminalSettings = GetCurrentTerminalSettings();
+				EnableTerminalRawMode();
+			}
+
+			var passwordBuilder = new StringBuilder();
 			while( true ) {
-				ConsoleKeyInfo input = Console.ReadKey( intercept: true );
-				if( input.Key == ConsoleKey.Enter ) {
-					Console.Error.Write( '\n' );
-					break;
+				char key = readKey();
+
+				if( key == CTRL_C ) {
+					throw new BmxException( "" );
```

Comment:
Seems a little naughty to execute the finally block by throwing a fake exception

Reply:
I wouldn't say it's a fake exception though, it actually ends the program. Putting cleanup inside `finally` is pretty standard.

I put an empty string here because this string is displayed to the user and I don't think the user needs to see anything when they themselves pressed Ctrl+C. Maybe a comment there would make sense.